compiler-tests/src/test/data/box/reports/ReallyLongPackageNameHandledWhenWritingDebugReports.kt

```diff
@@ -0,0 +1,62 @@
+// https://github.com/ZacSweers/metro/issues/1303
+/**
+ * Covers the case where debug report file names exceed the file name limit when we have the combination of long package and long name, e.g.
+ * java.nio.file.FileSystemException: /private/var/folders/wv/bkgjhyrj3l5f8pwkh_kfds000000gn/T/dev.zacsweers.metro.compiler.BoxTestGenerated$Reports2testReallyLongPackageNameHandledWhenWritingDebugReports/metro/reports/provider-factory-anvil.register.featureflags.com.squareup.feature.featureflags.RealProductionLoremIpsumFunctionalityFeatureFlagWrapper_AllowLoremIpsumFunctionalitySemiLongNameFeatureFlag.ProvidesAllowLoremIpsumFunctionalitySemiLongNameFeatureFlag$$MetroFactory.kt: File name too long
+ * 	< ... java internals stacktrace ... >
+ * 	at dev.zacsweers.metro.compiler.ir.IrMetroContextKt.writeDiagnostic(IrMetroContext.kt:265)
+ *
+ * 	In this example, the default report prefix and package name add up to almost 80 characters:
+ * 	provider-factory-anvil.register.featureflags.com.squareup.feature.featureflags
+ *
+ * 	This isn't exclusively a package issue, because it's still a combination of the related factory class name, but encoding the package as a report path segment instead of the file name is a simple way to significantly limit the cases where this happens without reducing generated code or file readability.
+ */
+// REPORTS_DESTINATION: metro/reports
```

Comment:
Would be cool if we could expose something in the test infra/base class to make this easy to grab